### ISSUE TITLE
Rule Change: `operator-assignment` should catch template interpolation

### ISSUE THREAD

**Issue by B2o5T** [eslint/eslint#15840]
### What rule do you want to change?

operator-assignment

### What change to do you want to make?

Generate more warnings

### How do you think the change should be implemented?

A new default behavior

### Example code

```js
foo = `${foo} baz ${qux}`
```

### What does the rule currently do for this code?

no warnings

### What will the rule do after it's changed?

foo += ` baz ${qux}`

### Participation

- [ ] I am willing to submit a pull request to implement this change.

### Additional comments

a real-world case - for example, I have `prefer-template` and `operator-assignment` enabled. For the following code 2 warnings were reported by `prefer-template` and `operator-assignment` rules.

```js
foo = foo + ` baz ${qux}`
```

But after fixing the first one, the error report from `operator-assignment` is gone as he doesn't catch template interpolation.
